### PR TITLE
SHIFTWEEK: feat: add spec dir and move cp openapi spec file

### PR DESCRIPTION
**Pull request by eggfoobar**
Shift Week Project:

Created spec folder to hold our spec files for validating configs.
Installed our openapi spec file to spec folder.

/assign @ggiguash

### CODE REVIEW

File: packaging/rpm/microshift.spec [hunk 1]
```diff
@@ -184,14 +184,17 @@ install -d -m755 %{buildroot}/%{_unitdir}
...
 install -d -m755 %{buildroot}/%{_sysconfdir}/microshift/manifests
 install -d -m755 %{buildroot}/%{_sysconfdir}/microshift/manifests.d
+install -p -m644 cmd/generate-config/config/config-openapi-spec.json %{buildroot}%{_sysconfdir}/microshift/spec/config-openapi-spec.json
 install -p -m644 packaging/microshift/config.yaml %{buildroot}%{_sysconfdir}/microshift/config.yaml.default
 install -p -m644 packaging/microshift/lvmd.yaml %{buildroot}%{_sysconfdir}/microshift/lvmd.yaml.default
 install -p -m644 packaging/microshift/ovn.yaml %{buildroot}%{_sysconfdir}/microshift/ovn.yaml.default
 
 # /usr/lib/microshift manifest directories for other packages to add to
 install -d -m755 %{buildroot}/%{_prefix}/lib/microshift
+install -d -m755 %{buildroot}/%{_prefix}/lib/microshift/spec
```
Comment (ggiguash) [inline packaging/rpm/microshift.spec:197]: I think this type of read-only files are better suited with in the `%{buildroot}%{_datadir}/microshift` directory.

Comment (ggiguash) [reply packaging/rpm/microshift.spec:197]: We already have release*.json and greenboot functions there.

Comment (eggfoobar) [reply packaging/rpm/microshift.spec]: Sure thing Giga, updated. Quick clarification, did you still want to keep the `spec` folder?

**Pull request merged**